### DISCUSSION TITLE
search vs find

### DISCUSSION THREAD

**Discussion by badihi** [strapi/strapi#7478] [Legacy Topics]
Please clarify the difference between the two methods of `service`s, `search()` and `find()`.

I could not find anything in the doc (as usual! 🤕)

**Comment by lauriejim** [comment]
Hello! I suggest you to check the `query` page to have a better understanding of the search.
https://strapi.io/documentation/v3.x/concepts/queries.html#api-reference
The usage of `_q` params that will search not just in one field.

Then, referring to `I could not find anything in the doc (as usual! 🤕)`
Can you please give me more precise detail about what is missing, how we can improve it.
By saying that, it doesn't help to improve it.

Thanks for your time.

**Comment by badihi** [comment]
Hello there,

Look at the example in the documentation:

```
strapi
  .query('restaurant')
  .search({ _q: 'my search query', _limit: 10, _start: 20 });
```

or the description `Returns entries based on a search on all fields allowing it`.

Suppose someone needs to understand how to use `search()` method at his first experience with the platform. What does this example suggests him to do? What would be `my search query`? I'm sorry to say that, but this documentation is one of the poorest that I have ever seen.

Also about the main question, searching among multiple fields would be possible using `find()` method, too. Also using `_where` parameter building every complex query would be possible (I'm sorry to nag again! but it was also so hard for me to find the right usage of `_where` parameter in the documentation). So what would be the exact benefit of using `search()` method?
